### PR TITLE
Fix issue where cache updates would not cause rerenders in `useSuspenseQuery` while in strict mode

### LINKED ISSUE #10428: `useSuspenseQuery` does not update cache in React strict mode
When using `useSuspenseQuery` with `<React.StrictMode />`, the result does not properly update when the cache has been updated. This is not currently a problem in `useQuery`.

It appears the `QueryManager` `queries` property is empty, which is necessary in order for cache updates to broadcast to all queries. 

**Likely reason**

My theory is that this seems to be related to the new [`fetchOnFirstSubscribe`](https://github.com/apollographql/apollo-client/blob/07f8645edcc769b83cde9d3813ac2e6f2d95ade2/src/core/ObservableQuery.ts#L113) option that was introduced for [`useSuspenseQuery`](https://github.com/apollographql/apollo-client/blob/release-3.8/src/react/hooks/useSuspenseQuery.ts#L282) to avoid issuing a network request when the observable is first subscribed to. 

Without this option (the default for `useQuery`), the observable will [call `reobserve`](https://github.com/apollographql/apollo-client/blob/07f8645edcc769b83cde9d3813ac2e6f2d95ade2/src/core/ObservableQuery.ts#L142) as soon as the first subscription starts, this way a network request can be initiated. Snaking through the code, you'll see that `reobserve` calls [`fetch`](https://github.com/apollographql/apollo-client/blob/07f8645edcc769b83cde9d3813ac2e6f2d95ade2/src/core/ObservableQuery.ts#L694-L706), which in turns calls [`this.queryManager.setObservableQuery(this)`](https://github.com/apollographql/apollo-client/blob/release-3.8/src/core/ObservableQuery.ts#L700). Its this method that is responsible for registering queries in the `queries` property.

We need to ensure the `QueryManager.setObservableQuery` gets called in some form `useSuspenseQuery` to ensure the query is registered in `QueryManager` to ensure cache updates can be observed properly.

**Intended outcome:**

`useSuspenseQuery` properly updates when the cache has been updated.

**Actual outcome:**

No cache updates occur.

**How to reproduce the issue:**

I'm able to reproduce this in the Spotify demo. If possible, I'll see if I can create a small reproduction to demonstrate the issue.

**Versions**

`@apollo/client@v3.8.0-alpha.3`

### PR DESCRIPTION
**Pull request by jerelmiller**
Closes #10428

Fixes an issue where using `useSuspenseQuery` would not react to cache updates while using React's strict mode.

### Checklist:

- [x] If this PR contains changes to the library itself (not necessary for e.g. docs updates), please include a changeset (see [CONTRIBUTING.md](https://github.com/apollographql/apollo-client/blob/main/CONTRIBUTING.md#changesets))
- [x] If this PR is a new feature, please reference an issue where a consensus about the design was reached (not necessary for small changes)
- [x] Make sure all of the significant new logic is covered by tests

### CODE REVIEW

**Comment by jerelmiller** [comment]
I can confirm this PR fixes the issue using the snapshot release.

To test using the Spotify demo:
* Start playing a song
* Like/unlike the song in the playbar using the heart icon

When on the latest alpha version, you'll notice the hear icon is not properly updated to reflect the change made. The heart should toggle on/off when liking/unliking the track but this does not occur. Using the snapshot release, perform the same process again and you'll notice the heart icon is now properly updated.

File: src/react/hooks/__tests__/useSuspenseQuery.test.tsx [hunk 1]
```diff
@@ -1671,6 +1678,90 @@ describe('useSuspenseQuery', () => {
...
+      expect(renders.suspenseCount).toBe(2);
+    }
+  );
+
+  it.each<SuspenseQueryHookFetchPolicy>([
+    'cache-first',
+    'network-only',
+    'cache-and-network',
+  ])(
+    'response to cache updates in strict mode while using a "%s" fetch policy',
```
Comment (jerelmiller) [inline src/react/hooks/__tests__/useSuspenseQuery.test.tsx:1733]:
```suggestion
    'responds to cache updates in strict mode while using a "%s" fetch policy',
```

**Review by alessbell — APPROVED** [review]
I agree with the case being made for this workaround—which the comment lays out very clearly 😄—and the `strictMode` option will make it straightforward to add tests for these scenarios in the future 🎉 

As an aside, it looks like nearly all tests still pass with `strictMode: true`, with only the `suspenseCount` assertions needing adjusting for obvious reasons. `allows the client to be overridden` and `uses default variables from the client when none provided in options`, however, look like they would either a) need further consideration or b) _potentially_ be an indicator of a strict mode issue we should look at more closely - @jerelmiller just flagging these two tests since you surely have more insight here 👍

File: src/react/hooks/useSuspenseQuery.ts [hunk 2]
```diff
@@ -342,13 +346,37 @@ function useObservableQueryResult<TData>(observable: ObservableQuery<TData>) {
...
+        // not trigger the observable's subscription. This occurs due to the new
+        // `fetchOnFirstSubscribe` option introduced with `useSuspenseQuery`,
+        // which avoids issuing a network request (via `reobserve`) when calling
+        // `subscribe` on the observable. Unfortunately `reobserve` is required
+        // to put the observable back into a non-torn-down state, which is not
+        // called due to this option. Instead we try delaying calling subscribe
+        // for the first time by allowing React to run this effect, tear it down,
+        // then set it back up again before we resubscribe.
+        //
+        // Authors note (Jerel): This feels super hacky and I hate it, but this
```
[the remark this answers is not in the document]
Comment (jerelmiller) [reply src/react/hooks/useSuspenseQuery.ts:363]: Ya......................................................

**Comment by jerelmiller** [comment]
@alessbell good callout. I'm going to try those tests with a strict mode variant and ensure they work as expected.

**Comment by jerelmiller** [comment]
@alessbell I've added tests for the 2 you called out, each using strict mode: [7277274](https://github.com/apollographql/apollo-client/pull/10601/commits/727727463db9d4f0d89e25758a36f4e7f07d6a99) and [44985b8](https://github.com/apollographql/apollo-client/pull/10601/commits/44985b8be18ca7ac6ad58abacb2d274ce83639e2). Just to confirm, do these tests satisfy what you're looking to answer, or is there something further I can test?

**Comment by alessbell** [comment]
@jerelmiller thanks! Yes, that answers it perfectly 🙏

**Comment by jerelmiller** [comment]
arghhhhh. I forgot to create a changeset for this branch. I'll create a separate PR that links to this one.

**Pull request merged**